platform/openide.util/test/unit/src/org/openide/util/BaseUtilitiesTest.java

```diff
@@ -557,4 +567,141 @@ public void test_toPrimitiveArray_returnsArrayOfPrimitives_whenGivenArrayOfObjec
+            //good
+        }
+    }
+    //--------------------------------------------------------------------------
+    public void test_getClassName_returnsClassName_forProperInvocation() {
+        
+        assertEquals("java.lang.String", getClassName(String.class));
+        assertEquals("java.lang.String[]", getClassName(String[].class));
+    }
+    //--------------------------------------------------------------------------
+    public void test_getShortClassName_throwsNullPointer_whenGivenNullArgument() {
+        
+        try {
+            getShortClassName(null);
+            fail();
+        } catch (final NullPointerException e) {
+            //good
+        }
+    }
+    //--------------------------------------------------------------------------
+    public void test_getShortClassName_returnsClassName_forProperInvocation() {
+        
+        assertEquals("String", getShortClassName(String.class));
+        assertEquals("String[]", getShortClassName(String[].class));
+        assertEquals("UnicodeBlock", getShortClassName(Character.UnicodeBlock.class));
+    }
+    //--------------------------------------------------------------------------
+    public void test_pureCalssName_thworsNullPointer_whenGivenNullArgument() {
+        
+        try {
+            pureClassName(null);
+            fail();
+        } catch (final NullPointerException e) {
+            //good
+        }
+    }
+    //--------------------------------------------------------------------------
+    public void test_pureCalssName__returnsEmptyString_whenGivenEmptyString() {
+        
+        assertEquals("", pureClassName(""));
+    }
+    //--------------------------------------------------------------------------
+    public void test_pureCalssName__returnsBlankString_whenGivenBlankString() {
+        
+        assertEquals(" \t", pureClassName(" \t"));
+    }
+    //--------------------------------------------------------------------------
+    public void test_pureCalssName__returnsClassName_whenGivenFullyQualifiedClassName() {
+        
+        assertEquals("java.lang.String", pureClassName("java.lang.String"));
+        assertEquals("UnicodeBlock", pureClassName("java.lang.Character$UnicodeBlock"));
+    }
+    //--------------------------------------------------------------------------
+    // It seems wrapStringToArray is not used at all, maybe we could remove it all together
+    // https://commons.apache.org/proper/commons-text/ provides more or less the same functionality
+    //--------------------------------------------------------------------------
+    public void test_wrapStringToArray_throwsNullPointer_whenGivenNullArgument() {
+        
+        try {
+            wrapStringToArray(null, 11, BreakIterator.getLineInstance(), true);
+            fail();
+        } catch (final NullPointerException e) {
+            //good
+        }
+        try {
+            //this is slopy behavior - null pointer exception happens only
```

Comment:
Why is this "sloppy" behaviour? Is it because:
1. `BreakIterator.getLineInstance()` (which is locale dependent, by the way) is not working as you expect.
2. Or because `wrapStringToArray` is not working properly.
3. Or because `wrapStringToArray` is returning a result that is unexpected to you?

Reply:
This behavior is simply unexpected (I should have used "unexpected" instead of "sloppy").

When a method is called with a null value of a parameter which is not allowd to be null, the most common hevior is for method to throw exception before it executes it's logic. 

Meanwhile wrapStringToArray when given null proceeds with it's logic without performing "not null" check first . If the argument is not dereferenced, everything is fine. If the argument is dereferenced - it throws NullPointer. 

So the right thing would be to tighten the boahavior of this method and introduced "null check" at the beginning of 
wrapStringToArray . On the other hand it seams that this method is never used within the code base (I performed text search), so meybe it can be removed entirely.

Reply:
As far as unexpected results wrapStringToArray.
I found them strange so I'd like to someone more knowledgeable with the platform tell me, whether this is ok or have I found a bug.

Reply:
Why do you find them "strange"?